### ISSUE TITLE
Split and Organize Checkstyle grammar inputs

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#5105]
Split from https://github.com/checkstyle/checkstyle/issues/3891 ,

We should look over the [grammar's package](https://github.com/checkstyle/checkstyle/tree/master/src/test/resources/com/puppycrawl/tools/checkstyle/grammars) and figure out how we wish to organize them. Do we want the same organization as https://github.com/checkstyle/checkstyle/issues/3891 or go a different route.

The naming scheme should be controlled by a UT similar to the one in https://github.com/checkstyle/checkstyle/issues/3891 .

**Comment by romani** [comment]
I will approve this issue when we come to decision on how to make it. For now I do not know how make order there.
probably main question is ..... What for ?

**Comment by rnveach** [comment]
> What for ?

Because we organized everything else. It is like enabling a check over our code, fixing all it's violations, and than leaving one big area suppressed and undone because we ask "Why should we?". The reason is why we start all this organizing to begin with, to bring order to chaos.

One example of something we should change that pops out to me is that some files start with `expected` instead of `Expected` which we have organized all other inputs for.

**Comment by romani** [comment]
> Because we organized everything else

yes, but we know rules and how it should looks like. Here I do not see proposal. 

> some files start with expected instead of Expected which we have organized all other inputs for.

ok. please put this to issue description to make it clear what is required to be done in this issue.

**Comment by rnveach** [comment]
@ps-sp You have any comments on how this folder should be organized beyond what I have mentioned?

**Comment by ps-sp** [comment]
@rnveach 

1. I see file names like `Input$UTNAME...` in various places. Is it better for some reason to include UT names in the file name ? Because we already have the containing folder named after the UT which makes having UT name in the file look like somewhat redundant to me, please correct me if I am wrong.

2. We right now have filenames like `expected....Ast`.txt. I feel the usage of the suffix `Ast` is incorrect. Either we should use the suffix `ParseTree` or we can even use suffixes `JavaTree` and `JavadocTree` (or `JavaParseTree` and `JavadocParseTree`). 

3. Right now we have test input files relating to java grammar in `.../test/resources/.../grammars/` and the ones relating to javadoc grammar in `.../test/resources/.../grammars/javadoc`. I feel all the test input files for java grammar should be in `/test/resources/.../grammars/java`.

4. `htmlTags` and `javadocTags` can be renamed to `htmltags` and `javadoctags` so that packages names just have lowercase characters. I don't know if we should rather rename them to `html_tags` and `javadoc_tags` but it might not look that good aesthetically.

5. In `.../test/resources/.../grammars/javadoc/` we have some files which apparently can't be put in either `htmlTags` folder  or `javadocTags` folder so _may be_ we can create a folder called `misc` for such files. We can also check what input files in `htmlTags` or `javadocTags` do not belong there and should be transferred to `misc`

**Comment by rnveach** [comment]
> Is it better for some reason to include UT names in the file name ? 

For other places this is to ensure the file name is unique. New contributors get confused on the file to change when they are all named the same.
I don't think the grammar folder should follow this way right now.

>  I feel all the test input files for java grammar should be in /test/resources/.../grammars/java

The separation of Java and Javadoc sounds good to me.

> feel the usage of the suffix Ast is incorrect. Either we should use the suffix ParseTree or we can even use suffixes JavaTree and JavadocTree

I agree on `Ast` but I think I have just been removing these suffixes from other tests and not adding new ones. Splitting of folders from java/javadoc should be enough to not add this suffix for now.

> htmlTags and javadocTags can be renamed to htmltags and javadoctags 

yes, but are we sure nothing in `grammars` right now should be in one of those folders. Also what if an input has both html and javadoc tags?

> misc 

I'm not sure about this. Can you give examples of what isn't html or javadoc tag related?

**Comment by ps-sp** [comment]
>I agree on Ast but I think I have just been removing these suffixes from other tests and not adding new ones.

You mean that `Ast` suffix is acceptable to you ? If yes, I felt that `Ast` suffix was incorrect because our results are not abstract syntax trees but parse trees. Yes separation of java grammar eliminates need for suffixes like `javatree` and `javadoctree` but we shouldn't have `Ast` either for theoretical reasons. We can have suffix `Cst` for concrete syntax tree or  suffix `parse` or even no suffix at all.

>yes, but are we sure nothing in grammars right now should be in one of those folders. Also what if an input has both html and javadoc tags?

No, we need files in these folders. I said we should rename these folder names to all lowercase letters. Please correct me if I misunderstood your statement. And if an input belongs to both cases we can put it in misc perhaps ? 

>I'm not sure about this. Can you give examples of what isn't html or javadoc tag related?

[We already have files in `.../test/resources/.../grammars/javadoc/`](https://github.com/checkstyle/checkstyle/tree/master/src/test/resources/com/puppycrawl/tools/checkstyle/grammars/javadoc) which have not been put in either `javadocTags` or `htmlTags`, so these files, like input for `LEADING_ASTERISK`s or input for `CR` as newline in javadoc and so on can be put into `misc` folder.
